cuda/plan.sh

```diff
@@ -0,0 +1,306 @@
+pkg_version=9.2.148
+_driverver=396.37
+pkg_maintainer="The Habitat Maintainers <humans@habitat.sh>"
+pkg_license=('custom')
+pkg_source="https://developer.nvidia.com/compute/${pkg_name}/9.2/Prod2/local_installers/${pkg_name}_${pkg_version}_${_driverver}_linux"
+pkg_filename="${pkg_name}_${pkg_version}_${_driverver}_linux.run"
+pkg_shasum=f5454ec2cfdf6e02979ed2b1ebc18480d5dded2ef2279e9ce68a505056da8611
+pkg_upstream_url="https://developer.nvidia.com/cuda-zone"
+
+## NOTE: Much of this plan copies what Archlinux did to repackage cuda.
+##       ref: https://git.archlinux.org/svntogit/community.git/tree/trunk/PKGBUILD?h=packages/cuda
+
+pkg_deps=(
+  core/gcc-libs
+  core/glibc
+  core/ncurses
+  core/python2
+  core/jre8
+  core/coreutils
+  core/busybox-static
+)
+pkg_build_deps=(
+  core/gcc
+  core/make
+  core/patchelf
+  core/perl
+)
+
+pkg_bin_dirs=(bin)
+pkg_lib_dirs=(
+  lib64
+  lib64/stubs
+  nvvm/lib64
+)
+pkg_include_dirs=(include)
+
+do_before() {
+  if [[ ! -r /usr/bin/perl ]]; then
+    ln -sv "$(pkg_path_for perl)/bin/perl" /usr/bin/perl
+    _clean_perl=true
+  fi
+
+  if [[ ! -r /bin/rm ]]; then
+  ln -sv "$(pkg_path_for coreutils)/bin/rm" /bin/rm
+  _clean_rm=true
+  fi
+}
+
+do_unpack() {
+  pushd "${HAB_CACHE_SRC_PATH}" > /dev/null || return 1
+  sh "${pkg_filename}" --extract="${HAB_CACHE_SRC_PATH}/${pkg_dirname}"
+
+  pushd "${pkg_dirname}" > /dev/null || return 1
+
+  # shellcheck disable=SC2211
+  ./cuda-*.run --noexec --keep
+
+  # shellcheck disable=SC2211
+  ./cuda-samples*.run --noexec --keep
+
+  popd > /dev/null || return 1
+
+  popd > /dev/null || return 1
+}
+
+do_prepare() {
+  # path hacks
+  # 1rd sed line: sets right path to install man files
+  # 2rd sed line: hack to lie installer, now detect launch script by root
+  # 3rd sed line: sets right path in .desktop files and other .desktop stuff (warnings by desktop-file-validate)
+  sed -e "s|/usr/share|${pkg_prefix}/share|g" \
+      -e 's|can_add_for_all_users;|1;|g' \
+      -e 's|Terminal=No|Terminal=false|g' -e 's|ParallelComputing|ParallelComputing;|g' \
+      -i pkg/install-linux.pl
+
+  # use python2
+  _fix_interpreter_in_path "pkg" '*.py' core/python2 bin/python
+  _fix_interpreter_in_path "pkg" '*.py' core/coreutils bin/env
+}
+
+do_build() {
+  return 0
+}
+
+do_install() {
+  pushd pkg > /dev/null || return 1
+
+  export PERL5LIB=.
+  perl install-linux.pl -prefix="${pkg_prefix}" -noprompt -nosymlink
+  perl install-sdk-linux.pl -cudaprefix="${pkg_prefix}" -prefix="${pkg_prefix}/samples" -noprompt
+
+  # Hack we need because of glibc 2.26 (https://bugs.archlinux.org/task/55580)
+  # without which we couldn't compile anything at all.
+  # Super dirty hack. I really hope it doesn't break other stuff!
+  # Hopefully we can remove this for later version of cuda.
+  sed -i "1 i#define _BITS_FLOATN_H" "${pkg_prefix}/include/host_defines.h"
+
+  # Needs gcc7
+  ln -s "$(pkg_path_for core/gcc)/bin/gcc" "${pkg_prefix}/bin/gcc"
```

Comment:
These two lines are a bit concerning, as gcc isn't a run time dependency so they will be broken links any time someone installs the package.  Are they necessary to be part of the package, or is having gcc be in the pkg_deps sufficent?

Reply:
Yes.  You are right. I need to add these as a runtime dep. I’ll commit an updated plan soon.

Reply:
fixed